src/core/sys/posix/sys/types.d

```diff
@@ -192,14 +192,28 @@ else version (Darwin)
 }
 else version (FreeBSD)
 {
+    import core.sys.freebsd.config;
+
     // https://github.com/freebsd/freebsd/blob/master/sys/sys/_types.h
     alias long      blkcnt_t;
-    alias uint      blksize_t;
-    alias uint      dev_t;
+
+    static if (__FreeBSD_version >= 1200000)
+    {
+        alias ulong blksize_t;
```

Comment:
This is `__int32_t` on both FreeBSD 11 and FreeBSD 12. So, the `ulong` here is wrong, and the `uint` in the FreeBSD 11 version below is wrong. It should be `int` in both cases.

https://github.com/freebsd/freebsd/blob/release/12.2.0/sys/sys/_types.h#L40
https://github.com/freebsd/freebsd/blob/release/12.2.0/sys/sys/types.h#L80

https://github.com/freebsd/freebsd/blob/release/11.4.0/sys/sys/_types.h#L38
https://github.com/freebsd/freebsd/blob/release/11.4.0/sys/sys/types.h#L78

Reply:
This was from your PR I think. :-)